vxquery-core/src/main/javacc/xquery-grammar.jj

```diff
@@ -1999,8 +1999,9 @@ ASTNode JsonConstructor() :
 }
 {
     (
-        result = ObjectConstructor()
-        | result = ArrayConstructor()
+        result = ArrayConstructor()
+        | LOOKAHEAD(2) result = ObjectConstructor()
```

Comment:
Do we need the `LOOKAHEAD(2)` here? Shouldn't `{` be a different token than `{|`?